src/index.js

```diff
@@ -320,70 +467,74 @@ const initialize = async () => {
      */
 
     deployButton.onclick = async () => {
-      let contract;
       contractStatus.innerHTML = 'Deploying';
 
-      try {
-        contract = await piggybankFactory.deploy();
-        await contract.deployTransaction.wait();
-      } catch (error) {
-        contractStatus.innerHTML = 'Deployment Failed';
-        throw error;
-      }
-
-      if (contract.address === undefined) {
-        return;
-      }
+      await deployPiggyBankContract();
 
-      console.log(
-        `Contract mined! address: ${contract.address} transactionHash: ${contract.transactionHash}`,
-      );
       contractStatus.innerHTML = 'Deployed';
       depositButton.disabled = false;
       withdrawButton.disabled = false;
 
-      depositButton.onclick = async () => {
-        contractStatus.innerHTML = 'Deposit initiated';
-        const result = await contract.deposit({
-          from: accounts[0],
-          value: '0x3782dace9d900000',
-        });
-        console.log(result);
-        contractStatus.innerHTML = 'Deposit completed';
-      };
+      // Deposit button functionality with different library
+      if (libraryInUse === 'ethers') {
+        depositButton.onclick = async () => {
+          contractStatus.innerHTML = 'Deposit initiated';
+          const result = await piggyBankContract.deposit({
+            from: accounts[0],
+            value: '0x3782dace9d900000',
+          });
+          console.log(result);
+          contractStatus.innerHTML = 'Deposit completed';
+        };
+      } else {
+        depositButton.onclick = async () => {
+          contractStatus.innerHTML = 'Deposit initiated';
+          await piggyBankContract.methods
+            .deposit()
+            .send({
+              from: accounts[0],
+              value: '0x3782dace9d900000',
+            })
+            .then((receipt) => {
+              console.log(receipt);
+            });
+          contractStatus.innerHTML = 'Deposit completed';
```

Comment:
This should be moved in then() callback function instead of being here. 
Also, it would be nice to have catch on error and handle that as well (log to console and display status text with error message).

Reply:
Fixed in: 91555c04c8088ee472c87cc25843642994ec92b4